---
title: Introduction
author: miquel [dot] anglada [at] crg [dot] eu
date: 2021-02-17

font-size: 11pt
geometry: margin=1.0in
urlcolor: blue

output:
    html_document:
      self_contained: true
      toc: true
      toc_depth: 3
      number_sections: true
    md_document:
      variant: markdown_github
---
# Load data
To start playing around with the functions from these packages we will use the [`palmerpenguins` data set](https://allisonhorst.github.io/palmerpenguins/articles/intro.html).
This simple data set has both continuous and categorical variables that make it perfect for showcasing how different functions work.

![](images/palmerpenguins.png)

```{r, message=FALSE, warning=FALSE}
require(dplyr)
require(tidyr)

penguins_url = 'https://raw.githubusercontent.com/allisonhorst/palmerpenguins/master/inst/extdata/penguins.csv'
dat = read.csv(url(penguins_url), stringsAsFactors = TRUE)
dat = dat %>% drop_na()
head(dat)
```


# General plotting with `ggpubr`
```{r message=FALSE, warning=FALSE}
require(ggpubr)
require(ggplot2)
```

Personally, I am a big fan of `ggpubr` because it allows to make insightful plots quickly for exploration that in turn can be further customized thanks to being built on top of `ggplot2`.

These are useful links for using this package:
- plotting functions
- [palettes](http://rpkgs.datanovia.com/ggpubr/reference/get_palette.html)
- figure making


Next, we will try to answer different questions using this library and `ggplot2`.

## How many penguins of each species did we observe in total?
```{r}
ggpie(dat %>% count(species), x = 'n', fill = 'species')
```

## How many penguins of each species and sex did we observe across the different islands?
```{r}
ggbarplot(dat %>% count(species, sex, island), x = 'species', y = 'n', fill = 'sex', 
          label = TRUE, position = position_dodge(0.7), facet.by = 'island', palette = 'lancet')
```


## What are the distributions of flipper lengths considering penguin species, sex and islands of origin?
```{r, message=FALSE, warning=FALSE}
gghistogram(dat, x = 'flipper_length_mm', fill = 'sex', facet.by = c('species','island'))
```
Alternatively, we can use stripcharts charts:
```{r}
ggstripchart(dat, x = 'island', y = 'flipper_length_mm', color = 'sex', facet.by = 'species', alpha = 0.5, position = position_jitterdodge(), add = 'median_iqr', add.params = list(color='black', group='sex', size=0.2))
```

## Are the differences of body mass between sexes significant if we control for species and island?
```{r}
ggstripchart(dat, x = 'island', y = 'body_mass_g', color = 'sex', facet.by = 'species', alpha = 0.5, position = position_jitterdodge(), add = 'median_iqr', add.params = list(color='black', group='sex', size=0.2))+
   stat_compare_means(aes(color = sex), label = "p.signif", method = 'wilcox.test')
```

## What is the relationship between flipper length, body mass and bill length?
```{r}
ggscatter(dat, x = 'flipper_length_mm', y = 'body_mass_g', color = 'bill_length_mm', alpha = 0.5)
```

## Could we have sampling bias in the relationship between flipper length and body mass?
```{r}
ggscatter(dat %>% mutate(year=factor(year)), x = 'flipper_length_mm', y = 'body_mass_g', alpha = 0.5, color = 'year', ellipse = TRUE)
```

## What is the spearman correlation coefficient between body mass and flipper length?
```{r, message=FALSE, warning=FALSE}
ggscatter(dat %>% mutate(year=factor(year)), x = 'flipper_length_mm', y = 'body_mass_g', alpha = 0.5, color = 'year', 
          add = 'reg.line', conf.int = TRUE, 
          cor.coef = TRUE,
          cor.coeff.args = list(method = 'spearman', label.sep = '\n')) + 
   theme(aspect.ratio = 1)
```

## Create and save a figure
```{r, eval=FALSE}
fontsize = 6
labsize = 2

# overview number of observations of every sex across islands and species
p1 = ggbarplot(dat %>% count(species, sex, island), x = 'species', y = 'n', fill = 'sex', 
               label = TRUE, lab.size = labsize,
               position = position_dodge(0.7), facet.by = 'island', palette = 'lancet') + 
   ylim(NA, 68)

# sex-related body mass distributions across islands and species
p2 = ggstripchart(dat, x = 'island', y = 'body_mass_g', color = 'sex', facet.by = 'species', 
                  alpha = 0.5, position = position_jitterdodge(), add = 'median_iqr', 
                  add.params = list(color='black', group='sex', size=0.2),
                  palette = 'lancet')+
            stat_compare_means(aes(color = sex), label = "p.signif", method = 'wilcox.test', size = labsize)

# association of flipper length and body mass
p3 = ggscatter(dat %>% mutate(year=factor(year)), x = 'flipper_length_mm', y = 'body_mass_g', alpha = 0.5, color = 'year', 
          add = 'reg.line', conf.int = TRUE, 
          cor.coef = TRUE,
          cor.coeff.args = list(method = 'spearman', label.sep = '\n', size = labsize)) + 
   theme(aspect.ratio = 1)


p1p2 = ggarrange(p1 + theme_pubr(base_size = fontsize), p2 + theme_pubr(base_size = fontsize), ncol = 1, common.legend = TRUE)
fig = ggarrange(p1p2, p3 + theme_pubr(base_size = fontsize), widths = c(2,1), heights = c(2, 1), labels = 'AUTO')

# save
ggsave('myfig.pdf', fig, width = 15, height = 10, unit = 'cm')
```

![](myfig.pdf)

# Heatmaps with `pheatmap`
```{r}
require(pheatmap)
# we are only interested in numeric columns
cols_oi = c('bill_length_mm','bill_depth_mm','flipper_length_mm','body_mass_g')
rownames(dat) = 1:nrow(dat)
pheatmap(dat[,cols_oi], scale = 'column', show_rownames = FALSE, annotation_row = dat[,c('species','island'),drop=FALSE])
```

# Phylogenetic trees with `ggtree`
[book](https://guangchuangyu.github.io/ggtree-book/chapter-ggtree.html)

## Basic tree
```{r}
require(ggtree)
set.seed(100)
tree <- rtree(50)
ggtree(tree, layout="circular") + geom_tiplab()
```

## Tree and multiple sequence alignment
```{r, warning=FALSE, message=FALSE, fig.height=10}
require(seqinr)
require(ape)

# create tree from alignment
fasta_file = file.path(here::here(), 'data', 'raw', 'TP53-mammals-alignment-aa.fa')
aln = read.alignment(fasta_file, format = 'fasta', whole.header=TRUE)
D = dist.alignment(aln)
tree = njs(D)

# plot tree with MSA
tree_plot = ggtree(tree)
msaplot(tree_plot, fasta = fasta_file) + ggtitle('tree with MSA')
```

# References



# Session Info
```{r session info}
sessionInfo()
```