---
title: Introduction
author: miquel [dot] anglada [at] crg [dot] eu
date: 2021-02-17

font-size: 11pt
geometry: margin=1.0in
urlcolor: blue

output:
    html_document:
      self_contained: true
      toc: true
      toc_depth: 3
      number_sections: true
    md_document:
      variant: markdown_github
---

Personally, I am a big fan of `ggpubr` because it allows to make simple quick plots
and at the same time, as being built on top og `ggplot2`, it also allows to customize them
to a very high degree.

# General plotting with `ggpubr`

```{r load data and libraries, message=FALSE, warning=FALSE}
require(ggpubr)
require(dplyr)
require(tidyr)
require(pheatmap)

penguins_url = 'https://raw.githubusercontent.com/allisonhorst/palmerpenguins/master/inst/extdata/penguins.csv'
dat = read.csv(url(penguins_url), stringsAsFactors = TRUE)
dat = dat %>% drop_na()
head(dat)
```
## Bar plot
```{r}
ggbarplot(dat %>% count(species), x = 'species', y = 'n')
ggbarplot(dat %>% count(species), x = 'species', y = 'n', fill = 'species')
```

## Pie plot
```{r}
ggpie(dat %>% count(island), x = 'n', color = 'island')
```

## Histogram
```{r}
gghistogram(dat, x = 'species', stat = 'count', fill = 'species')
gghistogram(dat, x = 'bill_length_mm')
gghistogram(dat, x = 'bill_length_mm', fill = 'sex')
gghistogram(dat, x = 'bill_length_mm', fill = 'island')
gghistogram(dat, x = 'bill_length_mm', fill = 'species')
```
## Balloon plot
```{r}
tbl = as.data.frame(table(dat$species, dat$island))
ggballoonplot(tbl, fill = 'value')
```

## Box plot
```{r}
ggboxplot(dat, x = 'species', y = 'body_mass_g')
ggboxplot(dat, x = 'species', y = 'body_mass_g', fill = 'sex')
ggboxplot(dat, x = 'species', y = 'body_mass_g', fill = 'sex', facet.by = 'island')
ggboxplot(dat, x = 'island', y = 'body_mass_g', fill = 'sex', facet.by = 'species')
```
## Strip chart
```{r}
ggstripchart(dat, x = 'island', y = 'body_mass_g', fill = 'sex', facet.by = 'species')
ggstripchart(dat, x = 'island', y = 'body_mass_g', color = 'sex', facet.by = 'species', alpha = 0.5)
ggstripchart(dat, x = 'island', y = 'body_mass_g', color = 'sex', facet.by = 'species', alpha = 0.5, position = position_jitterdodge())
?ggstripchart
ggstripchart(dat, x = 'island', y = 'body_mass_g', color = 'sex', facet.by = 'species', alpha = 0.5, position = position_jitterdodge(), add = 'median_iqr')
ggstripchart(dat, x = 'island', y = 'body_mass_g', color = 'sex', facet.by = 'species', alpha = 0.5, position = position_jitterdodge(), add = 'median_iqr', add.params = list(color='black'))
ggstripchart(dat, x = 'island', y = 'body_mass_g', color = 'sex', facet.by = 'species', alpha = 0.5, position = position_jitterdodge(), add = 'median_iqr', add.params = list(color='black', group='sex', size=0.2))
```
## Dot plot
```{r}
ggdotplot(dat, x = 'island', y = 'body_mass_g', color = 'sex', facet.by = 'species', alpha = 0.5, position = position_jitterdodge(), add = 'median_iqr', add.params = list(color='black', group='sex', size=0.2))
```
## Violin plot
```{r}
ggviolin(dat, x = 'island', y = 'body_mass_g', color = 'sex', facet.by = 'species', alpha = 0.5, position = position_jitterdodge(), add = 'median_iqr', add.params = list(color='black', group='sex', size=0.2))
ggviolin(dat, x = 'island', y = 'body_mass_g', color = 'sex', facet.by = 'species', alpha = 0.5, add = 'median_iqr', add.params = list(color='black', group='sex', size=0.2))
ggviolin(dat, x = 'island', y = 'body_mass_g', fill = 'sex', facet.by = 'species', alpha = 0.5, add = 'median_iqr', add.params = list(color='black', group='sex', size=0.2))
```
## Scatter plot
```{r}
ggscatter(dat, x = 'flipper_length_mm', y = 'body_mass_g', alpha = 0.5)
ggscatter(dat, x = 'flipper_length_mm', y = 'body_mass_g', alpha = 0.5, color = 'island')
ggscatter(dat, x = 'flipper_length_mm', y = 'body_mass_g', alpha = 0.5, color = 'sex')
ggscatter(dat, x = 'flipper_length_mm', y = 'body_mass_g', alpha = 0.5, color = 'year')
ggscatter(dat, x = 'flipper_length_mm', y = 'body_mass_g', alpha = 0.5, color = 'year') + theme(legend.key.width=unit(1,"cm")) 
ggscatter(dat, x = 'flipper_length_mm', y = 'body_mass_g', alpha = 0.5, color = 'year') + theme(legend.key.width=unit(1,"cm")) 
ggscatter(dat %>% mutate(year=factor(year)), x = 'flipper_length_mm', y = 'body_mass_g', alpha = 0.5, color = 'year') + theme(legend.key.width=unit(1,"cm")) 
```

# Change palette
```{r}

```


# Add annotations
## Stats
```{r}
ggstripchart()
```

```{r}
ggscatter(df, x = "wt", y = "mpg",
   color = "black", shape = 21, size = 3, # Points color, shape and size
   add = "reg.line",  # Add regressin line
   add.params = list(color = "blue", fill = "lightgray"), # Customize reg. line
   conf.int = TRUE, # Add confidence interval
   cor.coef = TRUE, # Add correlation coefficient. see ?stat_cor
   cor.coeff.args = list(method = "pearson", label.x = 3, label.sep = "\n")
   )
```
## Labels
```{r}
```

# Other
- `ggparagraph`
- `ggsummarytable`

# Combine plots into figures
```{r}
```

# Save your figures
```{r}
```

# Heatmaps with `pheatmap`
```{r}
# we are only interested in numeric columns
cols_oi = c('bill_length_mm','bill_depth_mm','flipper_length_mm','body_mass_g')
pheatmap(dat[,cols_oi])

pheatmap(dat[,cols_oi], scale = 'column')
pheatmap(dat[,cols_oi], scale = 'column', show_rownames = FALSE)

# annotate a dimension
rownames(dat) = 1:nrow(dat)
pheatmap(dat[,cols_oi], scale = 'column', show_rownames = FALSE, annotation_row = dat[,'species',drop=FALSE])
pheatmap(dat[,cols_oi], scale = 'column', show_rownames = FALSE, annotation_row = dat[,c('species','island'),drop=FALSE])
```

# Phylogenetic tree with `ggtree`
[book](https://guangchuangyu.github.io/ggtree-book/chapter-ggtree.html)
```{r}
require(ggtree)
set.seed(100)
tree <- rtree(50)
ggtree(tree)
ggtree(tree, layout="slanted")
ggtree(tree, layout="slanted") + geom_tiplab()
```
```{r, fig.height=10}
require(seqinr)
require(ape)
# tree and MSA
fasta_file = file.path(here::here(), 'data', 'raw', 'TP53-mammals-alignment-aa.fa')
# create tree
#aln = read.alignment(fasta_file, format = 'fasta')
aln = read.alignment(fasta_file, format = 'fasta', whole.header=TRUE)
D = dist.alignment(aln)
#tree = nj(D)
tree = njs(D)
ggtree(tree)
ggtree(tree, ladderize = FALSE)
ggtree(tree, ladderize = FALSE) + 
  xlim(NA,0.5) +
  geom_tiplab(align=TRUE, linetype='dashed', linesize=.3)
ggtree(tree, ladderize = FALSE, layout = 'circular') + 
  xlim(NA,0.5) +
  geom_tiplab(align=TRUE, linetype='dashed', linesize=.3)
tree_plot = ggtree(tree, ladderize = FALSE)
# plot tree with MSA
msaplot(tree_plot, fasta = fasta_file)
```

# References

# Session Info
```{r session info}
sessionInfo()
```